crates/contract/tests/sandbox/vote.rs

```diff
@@ -853,3 +855,63 @@ async fn test_successful_resharing_after_cancellation_clears_cancelled_epoch_id(
 
     Ok(())
 }
+
+#[tokio::test]
+async fn vote_new_parameters_errors_if_new_participant_is_missing_valid_attestation() {
+    let (worker, contract, mut current_participant_accounts, _) = init_env_secp256k1(1).await;
+
+    let state: ProtocolContractState = contract.view("state").await.unwrap().json().unwrap();
+    let ProtocolContractState::Running(initial_running_state) = state else {
+        panic!("State is not running: {:#?}", state)
+    };
+
+    let initial_epoch_id = initial_running_state.keyset.epoch_id;
+    let existing_params = initial_running_state.parameters.clone();
+    let mut new_participants = existing_params.participants().clone();
+
+    // Add a new participant
+    let (new_account, new_account_id, new_participant_info) = {
+        let (mut new_accounts, participants) = gen_accounts(&worker, 1).await;
+        let (new_account_id, _, new_participant_info) =
+            participants.participants().first().unwrap().clone();
+        let new_account = new_accounts.pop().unwrap();
+        (new_account, new_account_id, new_participant_info)
```

Comment:
[bikeshedding] I wish `gen_accounts` were more user-friendly so we didn’t have to do all that conversion dance.